### PR TITLE
Armor CCC: Code Cleanup and Changes

### PR DESCRIPTION
**Pull request by Alterist**
Removes all equipping restrictions on armor
Removes all equipping restrictions on uniforms
Gives RO and Synth uniforms the proper parent (this gives them the same armor as marine uniforms )
Unifies the list of allowed items for suit storage (except for SG harness)
removes """"Lore"""" from the armor/uniform descriptions 
Changes a few of the armor stats to better fit it's description

### CODE REVIEW

File: code/modules/clothing/suits/marine_armor.dm [hunk 1]
```diff
@@ -275,17 +237,16 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
...
-	desc = "A modified and refashioned suit of M3 Pattern armor designed to be worn by the loader of a USCM vehicle crew. While the suit is a bit more encumbering to wear with the crewman uniform, it offers the loader a degree of protection that would otherwise not be enjoyed."
+	desc = "A modified and refashioned suit of M3 Pattern armor designed to be worn by vehicle crew. While the suit is a bit more encumbering to wear, it offers the loader a degree of protection that would otherwise not be enjoyed."
 	icon_state = "tanker"
-	uniform_restricted = list(/obj/item/clothing/under/marine/officer/tanker)
+	armor = list(melee = 50, bullet = 60, laser = 45, energy = 40, bomb = 40, bio = 15, rad = 15)
 
-	New()
+/obj/item/clothing/suit/storage/marine/tanker/New()
 		select_gamemode_skin(type)
 		..()
```
Comment (Rohesie) [inline code/modules/clothing/suits/marine_armor.dm]:
```suggestion
		return ..()
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 2]
```diff
@@ -402,16 +336,43 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
...
+	armor = list(melee = 50, bullet = 80, laser = 40, energy = 25, bomb = 30, bio = 0, rad = 0)
+
+/obj/item/clothing/suit/storage/marine/MP/RO
+	icon_state = "officer"
+	name = "\improper M3 pattern officer armor"
+	desc = "A well-crafted suit of M3 Pattern Armor typically found in the hands of higher-ranking officers. Useful for letting your men know who is in charge when taking to the field"
+
+/obj/item/clothing/suit/storage/marine/MP/RO/New()
+	select_gamemode_skin(/obj/item/clothing/suit/storage/marine/MP/RO)
+	..()
```
Comment (Rohesie) [inline code/modules/clothing/suits/marine_armor.dm]:
```suggestion
	return ..()
```

File: code/modules/clothing/under/marine_uniform.dm [hunk 3]
```diff
@@ -175,43 +175,45 @@
...
 /obj/item/clothing/under/marine/veteran/UPP
 	name = "\improper UPP fatigues"
-	desc = "A set of UPP fatigues, mass produced for the armed-forces of the Union of Progressive Peoples. A rare sight, especially in ICC zones. This particular set sports the dark drab pattern of the UPP 17th battalion, 'Smoldering Sons', operating in the sparse UPP frontier in the Anglo-Japanese arm."
+	desc = "A set of UPP fatigues, mass produced for the armed-forces of the Union of Progressive Peoples."
 	icon_state = "upp_uniform"
 	min_cold_protection_temperature = ICE_PLANET_min_cold_protection_temperature
-	has_sensor = 0
-	suit_restricted = list(/obj/item/clothing/suit/storage/faction/UPP, /obj/item/clothing/suit/storage/marine/smartgunner/UPP)
+
+	suit_restricted = list(/obj/item/clothing/suit/storage/marine/UPP, /obj/item/clothing/suit/storage/marine/smartgunner/UPP)
```
Comment (Rohesie) [inline code/modules/clothing/under/marine_uniform.dm]: Eliminate at will.

File: code/modules/clothing/under/marine_uniform.dm [hunk 4]
```diff
@@ -222,8 +224,8 @@
 	desc = "A set of loose fitting fatigues, perfect for an informal mercenary. Smells like gunpowder, apple pie, and covered in grease and sake stains."
 	icon_state = "freelancer_uniform"
 	min_cold_protection_temperature = ICE_PLANET_min_cold_protection_temperature
-	has_sensor = 0
-	suit_restricted = list(/obj/item/clothing/suit/storage/faction/freelancer)
+
+	suit_restricted = list(/obj/item/clothing/suit/storage/marine/freelancer)
```
Comment (Rohesie) [inline code/modules/clothing/under/marine_uniform.dm]: Eliminate at will.

File: code/modules/clothing/under/marine_uniform.dm [hunk 5]
```diff
@@ -69,59 +66,61 @@
...
-	icon_state = "johnny"
 
 /obj/item/clothing/under/marine/officer/logistics
 	name = "marine officer uniform"
-	desc = "A uniform worn by commissioned officers of the USCM. Do the corps proud. It has shards of light Kevlar to help protect against stabbing weapons, bullets, and shrapnel from explosions, a small EMF distributor to help null energy-based weapons, and a hazmat chemical filter weave to ward off biological and radiation hazards."
+	desc = "A uniform worn by commissioned officers of the USCM. Do the corps proud."
 	icon_state = "BO_jumpsuit"
 
+/obj/item/clothing/under/marine/officer/req
+	name = "requisition officer suit."
```
[the remark this answers is not in the document]
Comment (RandomColonist) [reply code/modules/clothing/under/marine_uniform.dm]: While you here, please. Already fixed on original CM.

File: code/modules/clothing/suits/marine_armor.dm [hunk 6]
```diff
@@ -490,196 +431,84 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
...
+	desc = "Standard body armor of the UPP military, the UM5 (Union Medium Mk 5) is a medium body armor, roughly on par with the M3 pattern body armor."
 	icon_state = "upp_armor"
 	slowdown = SLOWDOWN_ARMOR_MEDIUM
-	flags_armor_protection = UPPER_TORSO|LOWER_TORSO
+	flags_marine_armor = ARMOR_LAMP_OVERLAY
 	armor = list(melee = 60, bullet = 60, laser = 50, energy = 60, bomb = 40, bio = 10, rad = 10)
-	uniform_restricted = list(/obj/item/clothing/under/marine/veteran/UPP)
 
-/obj/item/clothing/suit/storage/faction/UPP/commando
+/obj/item/clothing/suit/storage/marine/UPP/commando
```
Comment (Rohesie) [inline code/modules/clothing/suits/marine_armor.dm]: Perhaps if you want to unify all clothes into a tag `military` might be better than `marine`.

Comment (ActualRory) [reply code/modules/clothing/suits/marine_armor.dm]: This would be greatly appreciated.

**Comment by Alterist** [comment]
No longer WIP, please remove tag

**Comment by Alterist** [comment]
this is done, ready for TM

**Pull request closed without merging**